### PR TITLE
Added CHANGE_SECTOR_TYPE RTS command

### PR DESCRIPTION
**Pull request by akaAgar**
Added a RTS command (syntax is CHANGE_SECTOR_TYPE TAG NEWTYPE). It will change a sector special type on the fly. Useful to make a sector airless or damaging, change colormap, etc.

If the new sector type (as defined in SECTORS.DDF) has a colormap or fog specified, they will be used and override the current sector's. Otherwise, current colormap and fog settings are kept.

### CODE REVIEW

File: source_files/edge/rad_act.cc [hunk 1]
```diff
@@ -903,6 +903,30 @@ void RAD_ActLightSector(rad_trigger_t *R, void *param)
...
+	for (i=0; i < numsectors; i++)
+	{
+		if (sectors[i].tag != t->tag) continue;
+
+		P_SectorChangeSpecial(sectors + i, t->typenum);
+
+		if (sectors[i].props.special && sectors[i].props.special->fog_color != RGB_NO_VALUE)
+		{
+			sectors[i].props.fog_color = sectors[i].props.special->fog_color;
+			sectors[i].props.fog_density = 0.01f * sectors[i].props.special->fog_density;
```
Comment (dashodanger) [inline source_files/edge/rad_act.cc:920]: When changing fog properties for a sector, it is also important to invalidate any existing fogwalls (the translucent "faux walls" that reflect the fog color and density of the adjacent sector). See the last portion of the RAD_ActFogSector function for a reference.

Comment (akaAgar) [reply source_files/edge/rad_act.cc:924]: Fixed. Added a few comments to RAD_ActChangeSectorType too.

**Comment by dashodanger** [comment]
We might have to look in p_setup.cc and see what all the program does to a sector on level load after pulling its special type

**Comment by akaAgar** [comment]
> This is not as easy as you think it is.
> For example: -Lighting effects (e.g. blinking) will need to be affected. -Sector physical stuff (e.g. viscosity) will need to be affected.
> As it is now, pretty much the only things affected would be damage, colourmap, and airless.

That was *somehow* on purpose. As there are already ways to change a sector's friction and light using ACTIVATE_LINETYPE, I thought copying them was not required. But you're right, this results in a hard-to-predict behavior where the user doesn't know what properties will or will not be copied from the sector special.

I'll update the PR so that friction and other effects are copied if the new sector type uses them (like it copies the fog only if the new sector type is foggy).

Maybe a better solution would be to forgo CHANGE_SECTOR_TYPE and add new RTS commands to change each parameter on its own (SET_SECTOR_AIRLESS, SET_SECTOR_DAMAGE...). Still, being able to copy a whole sector type is really convenient...

**Comment by LoboEire** [comment]
Suddenly making a sector a secret will also screw with the level counters and savegames.

It might be safer to have a few new RTS commands to change the individual properties rather than changing the whole sector to a new type.

**Comment by akaAgar** [comment]
You're right. You can close this PR, I'll create a new one with a series of individual commands.

**Comment by dashodanger** [comment]
Closing per request

**Pull request closed without merging**